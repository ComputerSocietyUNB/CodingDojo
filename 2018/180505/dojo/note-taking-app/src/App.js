import React, { Component } from 'react'
import Note from './Note'
import ShowNotes from './ShowNotes'

class App extends Component {
  state = {
    value: '',
    listNotes: ['potato', 'oioi']
  }

  handleChange = (event) => {
    this.setState({
      value: event.target.value,
    })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Note handleChange={this.handleChange} value={this.state.value} />
        <ShowNotes listNotes={this.state.listNotes} />
      </div>
    )
  }
}

export default App
