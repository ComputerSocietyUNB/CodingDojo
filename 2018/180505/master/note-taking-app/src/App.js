import React, { Component } from 'react'
import Note from './Note'
import NoteList from './NoteList'

class App extends Component {
  state = {
    noteList: [
      'my first note',
    ],
    current: 0,
  }

  getCurrentNote = () => {
    const { noteList, current } = this.state

    return noteList[current]
  }

  handleChange = (event) => {
    const { noteList, current } = this.state

    this.setState({
      noteList: [...noteList.slice(0, current), event.target.value, ...noteList.slice(current + 1)]
    })
  }

  removeNotes = (notesToBeRemoved) => {
    this.setState(state => ({
      noteList: state.noteList.filter((note, index) => !notesToBeRemoved.includes(index))
    }))
  }

  addNote = () => {
    this.setState(({ noteList }) => ({
      noteList: [...noteList, noteList.length]
    }))
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <div className="App-intro">
          <NoteList
            list={this.state.noteList}
            addNote={this.addNote}
            removeNotes={this.removeNotes}
          />
          <Note
            note={this.getCurrentNote()}
            onNoteEdit={this.handleChange}
          />
        </div>
      </div>
    )
  }
}

export default App
