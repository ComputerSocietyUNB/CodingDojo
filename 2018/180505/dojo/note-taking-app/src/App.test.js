import React from 'react'
import { render, Simulate } from 'react-testing-library'
import 'dom-testing-library/extend-expect'
import App from './App'

describe('App', () => {
  let app

  beforeEach(() => {
    app = render(<App />)
  })

  it('renders without crashing', () => {
    const { container, getByText } = app
    expect(getByText('Welcome to React')).toMatchSnapshot()
  })

  it('makes it possible to write', () => {
    const { getByAltText } = app
    expect(getByAltText('note')).toMatchSnapshot()
  })

  it('show input value on screen', () => {
    const { getByAltText, container } = app
    const note = getByAltText('note')
    
    expect(container.querySelector('.shownotes')).toMatchSnapshot()
    note.value = "potato"
    Simulate.change(note)

    expect(container.querySelector('.shownotes')).toHaveTextContent('potato')
  })



})
