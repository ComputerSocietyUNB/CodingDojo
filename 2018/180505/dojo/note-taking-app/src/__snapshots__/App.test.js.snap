// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App makes it possible to write 1`] = `
<input
  type="text"
  alt="note"
  value=""
/>
`;

exports[`App renders without crashing 1`] = `
<h1
  class="App-title"
>
  Welcome to React
</h1>
`;

exports[`App show input value on screen 1`] = `
<li
  class="shownotes"
>
  potato
</li>
`;
